editor/shader_create_dialog.cpp

```diff
@@ -31,15 +31,22 @@
 #include "shader_create_dialog.h"
 
 #include "core/config/project_settings.h"
+#include "core/string/string_builder.h"
 #include "editor/editor_file_dialog.h"
 #include "editor/editor_scale.h"
+#include "modules/modules_enabled.gen.h"
 #include "scene/resources/shader_include.h"
-#include "scene/resources/visual_shader.h"
 #include "servers/rendering/shader_types.h"
 
+#ifdef MODULE_VISUAL_SHADER_ENABLED
+#include "modules/visual_shader/visual_shader.h"
+#endif // MODULE_VISUAL_SHADER_ENABLED
+
 enum ShaderType {
 	SHADER_TYPE_TEXT,
+#ifdef MODULE_VISUAL_SHADER_ENABLED
 	SHADER_TYPE_VISUAL,
+#endif // MODULE_VISUAL_SHADER_ENABLED
```

Comment:
This is problematic as it may break compatibility. Does this `ShaderType` get saved to some scene/resource files?

Reply:
No, it's only used in the shader create dialog.

Reply:
But doesn't it remember the selected value for future runs? If so, the selected value might change (similarly to what happens when using C# in the script create dialog and then switching to a non-Mono build). It may not be a big deal though, unless you have `SHADER_TYPE_INC` selected in a build with VisualShader, and then when you open in a build without VisualShader this becomes `SHADER_TYPE_MAX` and throws an error.